pkg/gitproviders/provider_test.go

```diff
@@ -754,15 +866,41 @@ var _ = Describe("Test org deploy keys creation", func() {
 })
 
 var _ = Describe("helpers", func() {
-	DescribeTable("DetectGitProviderFromUrl", func(input string, expected GitProviderName) {
-		result, err := DetectGitProviderFromUrl(input)
+	DescribeTable("detectGitProviderFromUrl", func(input string, expected GitProviderName) {
+		result, err := detectGitProviderFromUrl(input)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(result).To(Equal(expected))
 	},
 		Entry("ssh+github", "ssh://git@github.com/weaveworks/weave-gitops.git", GitProviderGitHub),
 		Entry("ssh+gitlab", "ssh://git@gitlab.com/weaveworks/weave-gitops.git", GitProviderGitLab),
+		Entry("ssh+gitlab with subgroup", "git@gitlab.com:gogittest/sub-group/nginxsub.git", GitProviderGitLab),
+	)
+
+})
+
+var _ = Describe("get owner from url", func() {
+	DescribeTable("getOwnerFromUrl", func(normalizedUrl string, providerName GitProviderName, expected string) {
+		u, err := url.Parse(normalizedUrl)
+		Expect(err).NotTo(HaveOccurred())
+		result, err := getOwnerFromUrl(*u, providerName)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result).To(Equal(expected))
+	},
+		Entry("github", "ssh://git@github.com/weaveworks/weave-gitops.git", GitProviderGitHub, "weaveworks"),
+		Entry("gitlab", "ssh://git@gitlab.com/weaveworks/weave-gitops.git", GitProviderGitLab, "weaveworks"),
+		Entry("gitlab with subgroup", "ssh://git@gitlab.com/weaveworks/sub_group/weave-gitops.git", GitProviderGitLab, "weaveworks/sub_group"),
 	)
 
+	DescribeTable("getOwnerFromUrl fail paths", func(normalizedUrl string, providerName GitProviderName, expected string) {
+		u, err := url.Parse(normalizedUrl)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = getOwnerFromUrl(*u, providerName)
+		Expect(err).To(HaveOccurred())
```

Comment:
Please check the actual error here.  We can jump on a chat as to why.

Reply:
done